### PR TITLE
Update tuminer

### PR DESCRIPTION
**Pull request by monsieurzhang**
1. add margin option
2. update docker file
3. support same lang for mine mode

### CODE REVIEW

**Comment by guillaumekln** [comment]
PyTorch is now publishing official Docker images: https://hub.docker.com/r/pytorch/pytorch/tags?page=1&ordering=last_updated

Could you check if they can help simplifying the Dockerfile?

File: utilities/tuminer/Dockerfile [hunk 1]
```diff
@@ -1,86 +1,22 @@
...
+      g++ wget cpio \
+      libgtest-dev swig3.0 \
+      libopenblas-dev \
+      git \
+      libssl-dev \
+      wget && \
+    git clone https://github.com/Kitware/CMake ~/cmake && \
+    cd ~/cmake && \
+    ./bootstrap && \
+    make -j 8 install
```
Comment (guillaumekln) [inline utilities/tuminer/Dockerfile:19]: Is it possible to install CMake with the package manager? `agt-get install cmake`

Comment (monsieurzhang) [reply utilities/tuminer/Dockerfile:19]: the default version installed by apt-get is lower than requirement.
I'm not sure how to update.

Comment (guillaumekln) [reply utilities/tuminer/Dockerfile:19]: Ok, so let's keep that. Just for reference, what is the required version?

Comment (monsieurzhang) [reply utilities/tuminer/Dockerfile:19]: CMake 3.17 or higher is required.  You are running version 3.10.2
For https://github.com/facebookresearch/faiss.git

**Pull request merged**